### PR TITLE
Integrate causes from Charity Nav API into Charity objects

### PR DESCRIPTION
**Pull request by ryan-draper**
Integrate causes (which are sub-categories within tags) from Charity Nav API into Charity objects for more specificity on trending and personalization.

### CODE REVIEW

**Comment by charu0291** [comment]
please give proper details in the PR description.

File: Donation-platform/src/main/java/com/google/charities/AddCharitiesFromJSON.java [hunk 1]
```diff
@@ -85,6 +129,51 @@
           put(RESEARCH, 20);
         }
     };
+
+  // HARDCODED MAPPING OF CAUSES TO TRENDING SCORES
+  public static final Map<String, Integer> causeScores =
+    new HashMap<String, Integer>() {
+        {
+          put(ADV, 15);
```
Comment (charu0291) [inline Donation-platform/src/main/java/com/google/charities/AddCharitiesFromJSON.java:137]: why dont we inititalize this map with 0? Are we updating this score later based on google trends API?

File: Donation-platform/src/main/java/com/google/charities/AddCharitiesFromJSON.java [hunk 2]
```diff
@@ -148,7 +248,9 @@ public void addCharities() {
             clone.getLink(),
```
Comment (charu0291) [inline Donation-platform/src/main/java/com/google/charities/AddCharitiesFromJSON.java:248]: i think I asked this before, what is clonecharity class for?

File: Donation-platform/src/main/java/com/google/model/Cause.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+  public String getName() {
+    return name;
+  }
+
+  public void setName(String name) {
+    this.name = name;
+  }
+
+  public Double getTrendingScoreCause() {
```
Comment (charu0291) [inline Donation-platform/src/main/java/com/google/model/Cause.java:69]: getCauseTrendingScore or getTrendingScoreForCause is more meaningful

File: Donation-platform/src/main/java/com/google/model/Cause.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    return trendingScore;
+  }
+
+  public void setTrendingScoreCause(Double trendingScore) {
```
Comment (charu0291) [inline Donation-platform/src/main/java/com/google/model/Cause.java:73]: ditto

File: Donation-platform/src/main/java/com/google/charities/AddCharitiesFromJSON.java (hunk 1 above)
Comment (ryan-draper) [reply Donation-platform/src/main/java/com/google/charities/AddCharitiesFromJSON.java]: Done. Set to zero.

File: Donation-platform/src/main/java/com/google/model/Cause.java (hunk 3 above)
Comment (ryan-draper) [reply Donation-platform/src/main/java/com/google/model/Cause.java]: Done.

**Pull request merged**